aptos-move/framework/aptos-stdlib/sources/cryptography/curves.move

```diff
@@ -0,0 +1,217 @@
+module aptos_std::curves {
+    use aptos_std::type_info::type_of;
+
+    // Structs and consts.
+
+    // Fake structs representing group type.
+    struct BLS12_381_G1 {}
+    struct BLS12_381_G2 {}
+    struct BLS12_381_Gt {}
+
+    //TODO: handle as u8 temporarily. Upgrade to u64.
+    struct Scalar<phantom Group> has drop {
+        handle: u8
+    }
+
+    struct Point<phantom Group> has drop {
+        handle: u8
+    }
+
+    /// Get internal handle for a Scalar. Currently needed by groth16 module.
+    /// TODO: can this be avoided?
+    public fun get_scalar_handle<Group>(s: &Scalar<Group>): u8 {
```

Comment:
If we have to use this, which I hope we can avoid, let's use `public(friend)` and make Groth16 a friend of this module.